### DISCUSSION TITLE
Where can I find Directus' migration scripts? And what is the best strategy to keep schemas sync in different envs?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#11012] [Archive]
Hi mods~

I am wondering where can I find Directus' migration scripts for MySQL DB? 
I mean those run after executing `npx directus database migrate:latest`

My company run Directus in different envs. To keep the Directus schemas (directus_collections, directus_fields, etc) sync between them, we create these tables by ourselves in SQL scripts instead of running `npx directus database migrate:latest`. 

The problem is that new versions sometimes make breaking changes to schemas, so we have to update our table creation scripts accordingly, which is why I would like to have access to the migration scripts, otherwise I would have to compare our current DB with the one created by newest Directus version manually to figure out what have changed, which is tedious.

Thanks!

**Comment by oreilles** [comment]
All migrations are located here: https://github.com/directus/directus/tree/main/api/src/database/migrations

**Reply by powerseed** [reply]
@Oreilles 
Thanks for the reply.
However, the migration files are not written in mysql statements, which is what my company uses.
Do you know there is any way to convert it into Mysql statements?
I could do it manually but it should not be the correct way to go down as it is tedious and time-consuming.
Thanks

**Reply by oreilles** [reply]
Not in a straightforward way, but you can refer to [the knex documentation](https://knexjs.org/) to understand what each statement do.
